frontend/src/state/chat-slice.ts

```diff
@@ -212,6 +212,17 @@ export const chatSlice = createSlice({
               content += `\n\n- ${host} (port ${port})`;
             }
           }
+          if (
+            recallObs.extras.custom_secrets_descriptions &&
+            Object.keys(recallObs.extras.custom_secrets_descriptions).length > 0
+          ) {
+            content += `\n\n**Custom Secrets**`;
+            for (const [name, description] of Object.entries(
+              recallObs.extras.custom_secrets_descriptions,
+            )) {
+              content += `\n\n- $${name}: ${description}`;
+            }
+          }
```

Comment:
This is an existing pattern so not blocking but I really dislike all the structured data in the recallObservation--IMO this should be much more generic. It should just have markdown we can display instead of having to hard code all this

Future work!

Reply:
Yeah def agree, I was scratching my head as to why custom secrets weren't showing only to find out I have to hard code it